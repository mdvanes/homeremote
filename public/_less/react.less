// for /react test

// Toggle
.btn-icon {
    font-size: 350%;
}

// Button-Set
.btn-group-justified {
    > a,
    > span {
        font-size: 230%;
    }
}

.margin-top {
    margin-top: 1em;
}

.log {
    color: #fff;
    height: 300px;
}