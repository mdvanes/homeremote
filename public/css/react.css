.btn-icon {
  font-size: 350%;
}
.btn-group-justified > a,
.btn-group-justified > span {
  font-size: 230%;
}
.margin-top {
  margin-top: 1em;
}
.log {
  color: #fff;
  height: 300px;
}
