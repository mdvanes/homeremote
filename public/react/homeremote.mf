CACHE MANIFEST
index.html
../library/bootstrap/css/bootstrap.min.css
../library/bootstrap/fonts/glyphicons-halflings-regular.woff2
../library/bootstrap/fonts/glyphicons-halflings-regular.woff
../library/bootstrap/fonts/glyphicons-halflings-regular.ttf
../css/react.css
build/bundle.js

NETWORK:
*
