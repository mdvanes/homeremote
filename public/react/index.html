<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>React</title>
    
    
  </head>
  <body>
    
    <script src="build/bundle.js"></script>
    
  </body>
</html>
