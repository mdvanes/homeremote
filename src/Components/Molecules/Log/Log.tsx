import React, { FC } from "react";
import { Card, CardContent, CardActions, Typography, Button } from "@material-ui/core";
// TODO
// import ClearLogButton from '../containers/ClearLogButton';
// import RadioInfoButton from '../containers/RadioInfoButton';
import gitinfoJson from "../../../gitinfo.json";
import packageJson from "../../../../package.json";

import "./Log.scss";
import { useSelector } from "react-redux";
import { RootState } from "../../../Reducers";
import { LogState, Logline } from "./logSlice";

export type Props = {
    loglines: Logline[];
};

const Log: FC<Props> = () => {
    const loglines = useSelector<RootState, LogState["lines"]>(
        (state: RootState) => state.loglinesNew.lines
    );
    return (
        <Card className="log-card">
            <CardContent className="log-card-text version">
                <Typography>
                    version: {gitinfoJson.hash}-{gitinfoJson.branch}-
                    {packageJson.version}
                </Typography>
            </CardContent>
            <CardContent className="log-card-text">
                {loglines &&
                    loglines.map((logline, index) => (
                        <div key={index}>{logline.message}</div>
                    ))}
            </CardContent>
            <CardActions className="log-card-actions">
                <Button>clear</Button>
                {/*<ClearLogButton />
                <RadioInfoButton />*/}
            </CardActions>
        </Card>
    );
};

export default Log;
