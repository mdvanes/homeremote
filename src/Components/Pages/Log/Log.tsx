import React, { FC } from "react";
import LogContainer from "../../../Containers/LogContainer";

const Log: FC = () => (
    <div className="col-xs-12 col-md-3 margin-top">
        <LogContainer />
    </div>
);

export default Log;
