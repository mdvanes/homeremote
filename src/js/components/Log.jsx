import React, { PropTypes } from 'react';
import {Card, CardText, CardActions} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import ClearLogButton from '../containers/ClearLogButton';

import './log.scss';

// TODO Log from outside this scope, e.g. from button-group.jsx
// Should make containers/button-group-container.jsx and dispatch logInfo there

class Log extends React.Component {
    constructor(props) {
        super(props);
        this.getInfo = this.getInfo.bind(this);
        this.getShellStatus = this.getShellStatus.bind(this);
        this.getShellStatus();
    }

    // TODO Extract to container like ClearLogButton
    getInfo() {
        fetch(this.props.infoUrl, {
            credentials: 'same-origin',
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
        })
        .then(data => data.json())
        .then(data => {
            const message = data.status;
            this.props.logInfo(message);
        })
        .catch(error => this.props.logError('error on get info: ' + error));
    }

    // TODO extract to separate file?
    getShellStatus() {
        fetch('/shell/status', {
            credentials: 'same-origin',
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
        })
        .then(data => data.json())
        .then(data => {
            if(data.status === 'ok') {
                const resultString = data.entries.map(entry => {
                    return `${entry.name} ${entry.result}`;
                }).join(' / ');
                this.props.logInfo(`getShellStatus: ${resultString}`);
            } else {
                throw new Error('getShellStatus failed');
            }
        })
        .catch(error => this.props.logError('error on getShellStatus: ' + error));
    }

    render() {
        return (
            <Card className="log-card">
                <CardText className="log-card-text">
                    {this.props.loglines.map((logline, index) =>
                        <div key={index}>
                            {logline.message}
                        </div>
                    )}
                </CardText>
                <div className="log"></div>
                <CardActions className="log-card-actions">
                    <ClearLogButton/>
                    <FlatButton onClick={this.getInfo} label="radio info"/>
                </CardActions>
            </Card>
        );
    }
}

Log.propTypes = {
    loglines: PropTypes.arrayOf(PropTypes.shape({
        message: PropTypes.string.isRequired
    }).isRequired).isRequired
};

export default Log;