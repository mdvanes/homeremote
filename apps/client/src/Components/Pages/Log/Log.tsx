import { FC } from "react";
import LogCard from "../../Molecules/LogCard/LogCard";

const Log: FC = () => <LogCard />;

export default Log;
