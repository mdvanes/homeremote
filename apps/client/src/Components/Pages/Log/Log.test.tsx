import { render } from "@testing-library/react";
import Log from "./Log";

jest.mock("../../Molecules/LogCard/LogCard", () => "mock-log-card");

describe("Log page", () => {
    it("contains the LogCard", () => {
        const { baseElement } = render(<Log />);
        expect(baseElement).toMatchSnapshot();
    });
});
