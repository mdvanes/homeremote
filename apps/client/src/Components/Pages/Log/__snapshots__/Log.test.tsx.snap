// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log page contains the LogCard 1`] = `
<body>
  <div>
    <mock-log-card />
  </div>
</body>
`;
